### PR TITLE
Add markdown documentation for the pydeck library

### PR DESCRIPTION
**Pull request by ajduberstein**
For #2929 

#### Background
Provide markdown docs for pydeck's core API

#### Change List
- Add markdown files describing the functionality of publicly exposed pydeck classes and functions

### CODE REVIEW

File: bindings/python/pydeck/docs/data_utils.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+def autocompute_viewport(
+    points,
+    view_proportion=1,
+    viewport_type=pydeck.ViewState)
+```
+
+Automatically computes a zoom level for the points passed in.
+
+#### Parameters
+`points` : `list` of `list` of `float` or `pandas.core.frame`
```
Comment (Pessimistress) [inline bindings/python/pydeck/docs/data_utils.md]: Use list

File: bindings/python/pydeck/docs/data_utils.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    A list of points
+`view_propotion` : `float`
+    Proportion of the data that is meaningful to plot
```
Comment (ibgreen) [inline bindings/python/pydeck/docs/data_utils.md]: I don't understand what this param does based on this description, but maybe that is OK...

File: bindings/python/pydeck/docs/data_utils.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+## pydeck.data\_utils.autocompute\_viewport
+
+```python
+def autocompute_viewport(
+    points,
+    view_proportion=1,
+    viewport_type=pydeck.ViewState)
+```
+
+Automatically computes a zoom level for the points passed in.
```
Comment (ibgreen) [inline bindings/python/pydeck/docs/data_utils.md]: Nits:
- "Computes a view state (long/lat and zoom level)"?
- Is the `auto` in `autocompute` justified, seeing as this function is explicitly and not automatically called?

File: bindings/python/pydeck/docs/data_utils.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+#### Returns
+    `pydeck.Viewport` : Viewport fitted to the data
+
+
+## pydeck.data\_utils.assign\_random\_colors
+
+Produces a lookup table keyed by each class of data, with value as an RGB array
+
+This helps enable multi-class visualization in pydeck
```
Comment (ibgreen) [inline bindings/python/pydeck/docs/data_utils.md]: A bit hard to understand exactly what this does, maybe an example snippet would help?

File: bindings/python/pydeck/docs/deck.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+`map_style` : `str`, default `"mapbox://styles/mapbox/dark-v9"`
+    URI for Mapbox basemap style
+
+`mapbox_key` : `str`, default None
+    Read on initialization from the MAPBOX_API_KEY environment variable. Defaults to None if not set.
+    See https://docs.mapbox.com/help/how-mapbox-works/access-tokens/#mapbox-account-dashboard
+
+`initial_view_state` : `pydeck.ViewState`, default `pydeck.ViewState()`
+    Initial camera angle relative to the map, defaults to a fully zoomed out 0, 0-centered map
```
Comment (ibgreen) [inline bindings/python/pydeck/docs/deck.md]: initial camera position, direction and zoom level?

File: bindings/python/pydeck/docs/deck.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+`layers` : `pydeck.Layer` or list of pydeck.Layer`, default `[]`
+    pydeck.Layer objects to render
+
+`views` : `list` of `pydeck.View`, default `[pydeck.View()]`
+    List of `pydeck.View` objects to render. If rendering a standard map, there is rarely a reason to modify this.
+
+`map_style` : `str`, default `"mapbox://styles/mapbox/dark-v9"`
+    URI for Mapbox basemap style
+
+`mapbox_key` : `str`, default None
```
Comment (Pessimistress) [inline bindings/python/pydeck/docs/deck.md]: - Ideally the param names can align with the standalone bundle API (`mapboxApiAccessToken`)
- How do you disable map?

**Pull request merged**